pkg/asset/installconfig/azure/validation.go

```diff
@@ -158,6 +211,29 @@ func validateMininumRequirements(fieldPath *field.Path, req resourceRequirements
 	return allErrs
 }
 
+func validateSecurityType(fieldPath *field.Path, securityType aztypes.SecurityTypes, instanceType string, capabilities map[string]string) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	_, hasTrustedLaunchDisabled := capabilities["TrustedLaunchDisabled"]
+	confidentialComputingType, hasConfidentialComputingType := capabilities["ConfidentialComputingType"]
+	isConfidentialComputingTypeSNP := confidentialComputingType == "SNP"
+
+	supportedSecurityType := true
+	switch securityType {
+	case aztypes.SecurityTypesConfidentialVM:
+		supportedSecurityType = hasConfidentialComputingType && isConfidentialComputingTypeSNP
+	case aztypes.SecurityTypesTrustedLaunch:
+		supportedSecurityType = !(hasTrustedLaunchDisabled || hasConfidentialComputingType)
+	}
+
+	if !supportedSecurityType {
+		errMsg := fmt.Sprintf("this security type is not supported for instance type %s", instanceType)
```

Comment:
Is it possible for you to formulate this errMsg within the case statement, so that the message can be tailored to to the unsupported case?

Reply:
Thank you for spotting this, I indeed left out the specifics from the unsupported case error message. Do you think [this](https://github.com/openshift/installer/pull/7469/files#diff-75606c90c997f4218689ce173a97316d61be39a8df4f5bebe08a4cd0e9f24973R229-R245) makes sense?